### ISSUE TITLE
Calling cancel on drag gets called twice (on strong drags)

### ISSUE THREAD

**Issue by morgs32** [react-spring/react-use-gesture#52]
See your codesandbox here: https://codesandbox.io/s/n9vo1my91p

If you give it a forceful drag, it will skip a slide because it appears the `cancel` method is getting called twice (it happens that the slide++ is embedded in the `cancel` call).

**Comment by morgs32** [comment]
In this line [here](https://github.com/react-spring/react-use-gesture/blob/master/index.js#L253), why do we have to use a `requestAnimationFrame`. Some other race condition?

**Comment by morgs32** [comment]
What the... I just "updated" codesandbox and this ceases to happen. Sorry for the noise.

**Comment by morgs32** [comment]
Ok... it only happens when dev tools is open?

**Comment by dbismut** [comment]
Hey @morgs32, thanks for reporting the issue, nice catch :) Yes, `requestAnimationFrame` is used to let the gesture logic finish before terminating it with `cancel`.

```js
onDrag = (canceled, cancel) => {
  if(!canceled) { // <-- logic that depends on calling cancel()
    console.log('calling cancel() with canceled equal to', canceled)
    cancel()
  }
  console.log('canceled is equal to', canceled)
}

/* output using RAF */
// calling cancel() with canceled equal to false
// canceled is equal to false
// canceled is equal to true

/* output without using RAF */
// calling cancel() with canceled equal to false
// canceled is equal to true
// canceled is equal to false <-- not exactly what we want
```

What seems to happen with the devtools opened in the example is that RAF debounces the call to terminate the gesture and the browser sometimes has time to execute the gesture twice, hence the logic (that depends on `cancel` being called) to increase the index occurring twice. I'll investigate further why this might happen, but one solution would be to remove RAF and it would be up to the user to use `cancel` as the final statement to its gesture logic:

 ```js
onDrag = (canceled, cancel) => {
  if(!canceled) {
    console.log('calling cancel() with canceled equal to', canceled)
    cancel()
  }
  else // <-- added else
    console.log('canceled is equal to', canceled)
}

/* output without using RAF */
// calling cancel() with canceled equal to false
// canceled is equal to true
```

**Comment by morgs32** [comment]
I figured as much! Here's what resolved it for me:
https://github.com/react-spring/react-use-gesture/pull/53

Of course, it was only really an issue because in development. Interesting!

**Comment by dbismut** [comment]
I think it's actually an issue every time the browser would drop frames so I believe this needs a fix! I'll review your PR now :)

**Comment by dbismut** [comment]
Should be fixed in `5.0.1`!
